dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -76,25 +74,6 @@ public sealed class Kernel : IKernel, IDisposable
         this._skillCollection = skillCollection;
     }
 
-    /// <inheritdoc/>
-    public ISKFunction RegisterSemanticFunction(string functionName, SemanticFunctionConfig functionConfig)
-    {
-        return this.RegisterSemanticFunction(SkillCollection.GlobalSkill, functionName, functionConfig);
-    }
-
-    /// <inheritdoc/>
-    public ISKFunction RegisterSemanticFunction(string skillName, string functionName, SemanticFunctionConfig functionConfig)
-    {
-        // Future-proofing the name not to contain special chars
-        Verify.ValidSkillName(skillName);
-        Verify.ValidFunctionName(functionName);
-
-        ISKFunction function = this.CreateSemanticFunction(skillName, functionName, functionConfig);
-        this._skillCollection.AddSemanticFunction(function);
-
-        return function;
-    }
-
     /// <inheritdoc/>
     public IDictionary<string, ISKFunction> ImportSkill(object skillInstance, string skillName = "")
```

Comment:
Next step would be to move remaining skills related methods - ImportSkill, ImportFunction to the place they naturally belong to SkillsCollection class since they have nothing to do with Kernel code – both, despite declared in Kernel class, just register functions in the Kernel skills collection. It'll simplify Kernel code a little bit.

After that, we can think about splitting SkillsCollection and group all code we have today for working with semantic, native, OpenAPI, gRPC skills, spread across bunch of Kernel extension methods today, to corresponding skill sets – SemanticSkillSet, NativeSkillSet, OpenApiSkillSet, and GrpcSkillSet. The grouping will allow to have better separation of concern since all logic related to particular type of skills will live in the skill set class or it’s extension methods and not in one common SkillsCollection one or Kernel extension methods.